src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/MapRequestPayloadValueResolver.php

```diff
@@ -0,0 +1,130 @@
+    ];
+
+    private const CONTEXT_DESERIALIZE = [
+        DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+    ];
+
+    public function __construct(
+        private readonly ?SerializerInterface $serializer,
+        private readonly ?DenormalizerInterface $denormalizer,
+        private readonly ?ValidatorInterface $validator,
+    ) {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $payloadMappers = [
+            MapQueryString::class => $this->mapQueryString(...),
+            MapRequestBody::class => $this->mapRequestBody(...),
+        ];
+
+        $payloads = [];
+        foreach ($payloadMappers as $mappingAttribute => $payloadMapper) {
+            $attributes = $argument->getAttributesOfType($mappingAttribute);
+
+            if (!$attributes) {
+                continue;
+            }
+
+            $type = $argument->getType();
+            if (!$type) {
+                throw new \LogicException(sprintf('Could not resolve the "$%s" controller argument: argument should be typed.', $argument->getName()));
+            }
+
+            $payload = $payloadMapper($request, $type, $attributes[0]);
+
+            if ($this->validator) {
+                $violations = $this->validator->validate($payload);
+
+                if (\count($violations)) {
+                    throw new ValidationFailedException($payload, $violations);
+                }
+            }
+
+            $payloads[] = $payload;
+        }
+
+        return $payloads;
+    }
+
+    private function mapQueryString(Request $request, string $type, MapQueryString $attribute): ?object
+    {
+        return $this->denormalize($request->query->all(), $type, $attribute->context);
+    }
+
+    private function mapRequestBody(Request $request, string $type, MapRequestBody $attribute): ?object
+    {
+        if ($request->request->all()) {
+            return $this->denormalize($request->request->all(), $type, $attribute->context);
+        }
+
+        $format = $request->getPreferredFormat(self::DEFAULT_FORMAT);
+
+        try {
+            return $this->deserialize($request->getContent(), $type, $format, $attribute->context);
+        } catch (NotEncodableValueException $exception) {
+            throw new HttpException(Response::HTTP_BAD_REQUEST, sprintf('Request payload contains not valid %s.', $format), $exception);
+        }
+    }
+
+    private function denormalize(array $data, string $type, array $context): ?object
+    {
+        if (!$this->denormalizer) {
+            throw new \LogicException(sprintf('The "symfony/serializer" component is required to use the "%s" attribute. Try running "composer require symfony/serializer".', MapQueryString::class));
+        }
+
+        $format = self::DEFAULT_FORMAT;
+
+        return $this->denormalizer->denormalize($data, $type, $format, $context + self::CONTEXT_DENORMALIZE);
```

Comment:
```suggestion
        return $this->denormalizer->denormalize($data, $type, self::DEFAULT_FORMAT, $context + self::CONTEXT_DENORMALIZE);
```

Reply:
It was at the beginning but in this case we have long line which should be re-formatted and split to multilines